lib/Page.js

```diff
@@ -71,6 +80,38 @@ Page.prototype.collectIncludedFiles = function (dependencies) {
   });
 };
 
+/**
+ * Records the front matter data of the current page into this.frontMatter
+ * @param includedPage a page with its dependencies included
+ */
+Page.prototype.collectFrontMatter = function (includedPage) {
+  return new Promise((resolve) => {
+    const $ = cheerio.load(includedPage);
+    const frontMatter = $('frontmatter');
+    const includedFrontMatter = $('frontmatter div');
+    if (frontMatter.get(FIRST_ELEMENT_INDEX) !== undefined) {
+      let frontMatterData = '';
+      if (includedFrontMatter.get(FIRST_ELEMENT_INDEX) !== undefined) {
+        frontMatterData = $(includedFrontMatter).get(FIRST_ELEMENT_INDEX).children[FIRST_ELEMENT_INDEX].data;
+      } else {
+        frontMatterData = frontMatter.get(FIRST_ELEMENT_INDEX).children[FIRST_ELEMENT_INDEX].data;
+      }
+      const formattedMatter = `${FRONT_MATTER_FENCE + NEW_LINE}${frontMatterData}${FRONT_MATTER_FENCE}`;
+      // Parse front matter data
+      const parsedData = fm(formattedMatter);
+      parsedData.attributes.src = this.pageSrc;
+      parsedData.attributes.title = this.pageTitlePrefix + parsedData.attributes.title;
+      this.frontMatter = parsedData.attributes;
+      this.title = this.frontMatter.title;
+      resolve(includedPage);
+    } else {
+      // Page is addressable but does not have a specified title
+      this.frontMatter = { src: this.pageSrc };
+      resolve(includedPage);
```

Comment:
We can combine line 110 and 106 into a line after the if else bllock

Reply:
Will fix 👍